### DISCUSSION TITLE
Do we guarantee that the contents of `buf` passed to `poll_write` of `AsyncWrite` are the same?

### DISCUSSION THREAD

**Discussion by RealXuChe** [tokio-rs/tokio#3990] [Q&A]
I'm implementing `AsyncWrite` for some struct. If `poll_write` returned `Pending` for the first poll, is it safe to suggest that the `buf` passed next time has the same content as the first-poll one?

Or, does the following `buf`'s content passed before returning `Ready` only appends the previous one?

(Sorry for my bad English)

**Comment by Darksonn** [comment]
No. This is not guaranteed.
